### PR TITLE
The allow_keys global is unused in nvim, remove it

### PR DESCRIPTION
**Pull request by hardenedapple**
It's only use (as opposed to being incremented and decremented) was removed in 10a5825b95c8072d7be5cc0ddb308b09605b6b6e

### CODE REVIEW

File: src/nvim/getchar.c [hunk 1]
```diff
@@ -1389,27 +1389,21 @@ int vgetc(void)
...
       }
       c = vgetorpeek(true);
       if (did_inc) {
-        --no_mapping;
-        --allow_keys;
+        no_mapping--;
       }
 
       /* Get two extra bytes for special keys */
       if (c == K_SPECIAL
```
Comment (bfredl) [inline src/nvim/getchar.c]: maybe join these lines while we're at it?

**Review by bfredl — APPROVED** [review]

File: src/nvim/getchar.c (hunk 1 above)
Comment (hardenedapple) [inline src/nvim/getchar.c]: Sorry, I don't follow. Join which lines?

Comment (bfredl) [inline src/nvim/getchar.c]: This one and the next.

Comment (hardenedapple) [inline src/nvim/getchar.c]: As in 1401 and 1402?  Sure :)

**Pull request merged**